packages/components/datepicker/src/Datepicker.tsx

```diff
@@ -1,27 +1,177 @@
-import React from 'react';
+import React, {
+  ChangeEventHandler,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
 import { cx } from 'emotion';
 import { CommonProps } from '@contentful/f36-core';
 
+import { format, isValid, parse, startOfDay, endOfDay } from 'date-fns';
 import { getStyles } from './Datepicker.styles';
 import { Calendar } from './Calendar';
+import { Popover } from '@contentful/f36-popover';
+import { DayPickerSingleProps } from 'react-day-picker';
+import FocusLock from 'react-focus-lock';
+import { TextInput, TextInputProps } from '@contentful/f36-forms';
+import { IconButton } from '@contentful/f36-button';
+import { CalendarIcon } from '@contentful/f36-icons';
+import { PopoverProps } from '@contentful/f36-popover/src';
 
-export type DatepickerProps = CommonProps;
+const DATE_FORMAT = 'dd/MM/yyyy';
 
-function _Datepicker(props: DatepickerProps, ref: React.Ref<HTMLDivElement>) {
+export type DatepickerProps = CommonProps & {
+  onSelect: (day: Date | undefined) => void;
+} & Omit<
+    DayPickerSingleProps,
+    'mode' | 'onSelect' | 'fromMonth' | 'toMonth' | 'fromYear' | 'toYear'
+  > &
+  Pick<
+    TextInputProps,
+    'id' | 'isDisabled' | 'isInvalid' | 'isReadOnly' | 'isRequired'
+  > &
+  Pick<PopoverProps, 'usePortal'>;
+
+/**
+ * Provides functionality for date selection.
+ */
+export function Datepicker(props: DatepickerProps) {
   const styles = getStyles();
+  const {
+    //common props
+    testId = 'cf-datepicker',
+    className,
+    style,
+    // input props
+    id,
+    isDisabled,
+    isInvalid,
+    isReadOnly,
+    isRequired,
+    // popover props
+    usePortal,
+    // calendar props
+    selected,
+    onSelect,
+    fromDate,
+    toDate,
+    locale,
+    ...restCalendarProps
```

Comment:
nit: we use `otherProps` across our components

Reply:
done